src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -397,8 +402,82 @@ def makeSubscriptions(self, workflow):
+                                    msg += "Dataset Names: %s"
+                                    self.logger.warning(msg, workflow['RequestName'], dMap['datasets'])
+                                    continue
+                                rseExpression = pnnExpr + '&cms_type=real&rse_type=DISK'
+                            else:
+                                rseExpression = '(tier=2|tier=1)&cms_type=real&rse_type=DISK'
+                            scope = 'cms'
+                            if self.msConfig['enableDataPlacement']:
+                                rucioResultList.append(self.rucio.createReplicationRule(dMap['datasets'],
+                                                                                        rseExpression,
+                                                                                        copies=copies,
+                                                                                        scope=scope))
+                            else:
+                                msg = "Dry run mode. The effective Rucio submission would look like: \n"
+                                msg += "account: %s \n"
+                                msg += "dids: %s \n"
+                                msg += "rseExpression: %s\n"
+                                msg += "copies: %s\n"
+                                self.logger.warning(msg,
+                                                    self.msConfig['rucioAccount'],
+                                                    pformat([{'scope': scope, 'name': did} for did in dMap['datasets']]),
+                                                    rseExpression,
+                                                    copies)
+                        except Exception as ex:
+                            msg = "Could not make transfer subscription for Workflow: %s\n:%s"
+                            self.logger.exception(msg, workflow['RequestName'], str(ex))
+                            return workflow
+                else:
+                    # NOTE:
+                    #    Nothing else to be done here. We mark the document as
+                    #    done so we do not iterate through it multiple times
+                    msg = "Skip submissions for %s. Either all data Tiers were "
+                    msg += "excluded or there were no Output Datasets at all."
+                    self.logger.warning(msg, workflow['RequestName'])
+                    self.docKeyUpdate(workflow, transferStatus='done')
+                    return workflow
 
+                transferIDs = rucioResultList
+                if self.msConfig['enableDataPlacement']:
+                    self.docKeyUpdate(workflow,
+                                      transferStatus='done',
+                                      transferIDs=transferIDs)
+                return workflow
+
+            elif isinstance(workflow, (list, set, CommandCursor)):
```

Comment:
I thought I had made a comment on this block if code. Can you please elaborate a bit here Todor? Is this another case that should never happened and we better go with raising an exception?

Reply:
No, Actually it is something that must happen. As we discussed during the previous weeks, since we are not anticipating for this service to to become multithreaded ever, the only plan for scaling this if needed was to start  feeding the `Consumer` thread instead of a single document at a time in a set of documents at a time. And since this function needs to change behavior based on the way it is called this is actually how it recognizes it's run mode.  

And I checked, but rising an exception is actually a bad idea - it breaks the pipeline and needs to be handled outside - which leads to not needed complications. As I previously said this, is not expected to change during runtime with some workflows or any change of running conditions. This is to distinguish the actual run mode - how  the pipeline/thread is feeded.

Reply:
And what does it mean having a `CommandCursor` object type? Is it a generator object yielding data straight from Mongo?

Instead of this `pass` statement, perhaps we should then add something like:
```
raise NotImplementedError("This block needs to be implemented to support bulk workflow operations")
```

Reply:
Yes this is  a generator coming from MongoDB. 
But I think rising such an error is an unneeded complication.  We just need to implement the code for working on a set of workflows.

Reply:
Upon a second thought, I see your point here and I like the idea. I am implementing the additional class and adding this exception here. Just to mention: this will break the whole module and would require someone to come and take care of the error. But since it should not happen at runtime, we may be safe.

Reply:
Done